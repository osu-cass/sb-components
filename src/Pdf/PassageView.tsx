import * as React from 'react';
import { ItemView } from './ItemView';
import { ItemPdfModel } from './PdfModels';

export interface PassageViewProps {
    view: ItemPdfModel;
    associatedItems: string[];
}

export class PassageView extends React.Component<PassageViewProps, {}> {
    render() {
        return (
            <div className='item'>
                <h2>Passage for {this.props.associatedItems.length === 1 ? 'item' : 'items'} {this.props.associatedItems.join(', ')}</h2>
                <ItemView view={this.props.view} />
            </div>
        );
    }
}