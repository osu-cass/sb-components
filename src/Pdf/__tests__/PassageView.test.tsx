import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { PassageView } from '../PassageView';
import { shallow, mount, render } from 'enzyme';
import { passageViewProps } from './PdfMocks';

describe("EvidenceStatement", () => {
    it("matches snapshot", () => {
        expect(shallow(<PassageView {...passageViewProps} />)).toMatchSnapshot()
    })
})