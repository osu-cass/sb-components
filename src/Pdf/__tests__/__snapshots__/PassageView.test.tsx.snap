// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvidenceStatement matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PassageView
    associatedItems={
      Array [
        "187-1234",
        "187-1235",
      ]
    }
    view={
      Object {
        "captured": true,
        "html": null,
        "id": "12334",
        "picturePath": null,
        "type": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Passage for 
          items
           
          187-1234, 187-1235
        </h2>,
        <ItemView
          view={
            Object {
              "captured": true,
              "html": null,
              "id": "12334",
              "picturePath": null,
              "type": 1,
            }
          }
        />,
      ],
      "className": "item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Passage for ",
            "items",
            " ",
            "187-1234, 187-1235",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Passage for ",
          "items",
          " ",
          "187-1234, 187-1235",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "view": Object {
            "captured": true,
            "html": null,
            "id": "12334",
            "picturePath": null,
            "type": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Passage for 
            items
             
            187-1234, 187-1235
          </h2>,
          <ItemView
            view={
              Object {
                "captured": true,
                "html": null,
                "id": "12334",
                "picturePath": null,
                "type": 1,
              }
            }
          />,
        ],
        "className": "item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Passage for ",
              "items",
              " ",
              "187-1234, 187-1235",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Passage for ",
            "items",
            " ",
            "187-1234, 187-1235",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "view": Object {
              "captured": true,
              "html": null,
              "id": "12334",
              "picturePath": null,
              "type": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
